includes/class-wcmp-frontend.php

```diff
@@ -11,548 +11,658 @@
+            if ( empty($myparcel_delivery_options_always_display) && !empty($chosen_shipping_methods) && is_array($chosen_shipping_methods) ) {
+                $shipping_country = WC()->customer->get_shipping_country();
+                if ($shipping_country != 'NL' && $shipping_country != 'BE') {
+                    $show_delivery_options = false;
+                }
+                $chosen_shipping_method = array_shift($chosen_shipping_methods);
+                $shipping_class = $this->get_cart_shipping_class();
+                $package_type = WooCommerce_MyParcel()->export->get_package_type_from_shipping_method( $chosen_shipping_method, $shipping_class, $shipping_country );
+                if ($package_type != 1) { // parcel
+                    $show_delivery_options = false;
+                }
+            }
+
+            // Don't load when cart doesn't need shipping
+            if ( false == WC()->cart->needs_shipping()) {
+                $show_delivery_options = false;
+            }
+
+            $delivery_options_shipping_methods = $this->getDeliveryOptionsShippingMethods();
+            $delivery_options_always_display = $this->myparcelDeliveryOptionsAlwaysDisplay()? 'yes' : 'no';
+
+            $urlJs       = WooCommerce_MyParcel()->plugin_url() . "/assets/delivery-options/js/myparcel.js";
+
+            $jsonConfig  = $this->get_checkout_config();
+
+            echo "<script>" .
+                 "myparcel_delivery_options_always_display = '$delivery_options_always_display';" .
+                 "myparcel_delivery_options_shipping_methods = $delivery_options_shipping_methods;" .
+                 "myParcelConfig = {$jsonConfig}; myparcel_delivery_options_shipping_methods = {$delivery_options_shipping_methods} </script>";
+            require_once(WooCommerce_MyParcel()->plugin_path().'/includes/views/wcmp-delivery-options-template.php');
+
+            return;
+
+        }
+
+
+        // XXX Move to Jquery ?
+        public function output_shipping_data() {
+            $shipping_data = $this->get_shipping_data();
+            printf('<div class="myparcel-shipping-data">%s</div>', $shipping_data);
+        }
+
+        // XXX Move to Jquery ?
+        public function get_shipping_data() {
+
+            if ($shipping_class = $this->get_cart_shipping_class()) {
+                $shipping_data = sprintf('<input type="hidden" value="%s" id="myparcel_highest_shipping_class" name="myparcel_highest_shipping_class">', $shipping_class);
+                return $shipping_data;
+            }
+
+            return false;
+        }
+
+        /**
+         * Save delivery options to order when used
+         *
+         * @param  int   $order_id
+         * @param  array $posted
+         *
+         * @return void
+         */
+
+        public function save_delivery_options( $order_id, $posted ) {
+            $order = WCX::get_order( $order_id );
+
+            if ($_POST['myparcel_highest_shipping_class'] != NULL ) {
```

Comment:
Eigenlijk moet je `get_post(...)` of `get_post_field` ofzo van wp gebruiken. Maar dit was al zo, dus laten we het maar zo.

Reply:
Oke, ik zal het noteren, misschien voor een volgende versie.